src/main/java/com/bite_sized_knowledge/bite_backend/global/filter/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,106 @@
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    private final JwtService jwtService;
+    private final MemberRepository memberRepository;
+
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+    private static final String ERROR_INVALID_TOKEN = "유효하지 않은 토큰입니다.";
+    private static final String ERROR_MEMBER_NOT_FOUND = "회원 정보가 존재하지 않습니다.";
+    private static final String ERROR_UNEXPECTED = "서버 내부 오류가 발생했습니다.";
+
+    @Override
+    protected void doFilterInternal(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            FilterChain filterChain
+    ) throws IOException, ServletException {
+
+        try {
+            String header = request.getHeader(AUTHORIZATION_HEADER);
+
+            if (isTokenPresentAndValid(header)) {
+                String token = extractToken(header);
+                if (!jwtService.isValidToken(token)) {
+                    throw new BadCredentialsException(ERROR_INVALID_TOKEN);
+                }
+
+                if (!processAuthentication(token, request)) {
+                    throw new UsernameNotFoundException(ERROR_MEMBER_NOT_FOUND);
+                }
+            }
+
+            // 유효 토큰이면 체인 진행
+            filterChain.doFilter(request, response);
+
+        } catch (BadCredentialsException | UsernameNotFoundException ex) {
+            // 인증 관련 예외
+            // Spring Security 상에서 AuthenticationEntryPoint가 처리하도록 예외 다시 던짐
+            throw ex;
+        } catch (Exception ex) {
+            // 나머지 예상치 못한 예외(서버 에러) -> 필요시 500 응답
+            throw new RuntimeException(ERROR_UNEXPECTED, ex);
+        }
+    }
+
+    private boolean isTokenPresentAndValid(String header) {
+        return StringUtils.hasText(header) && header.startsWith("Bearer ");
+    }
+
+    private String extractToken(String header) {
+        return header.substring(7); // "Bearer " 이후의 토큰만 추출
```

Comment:
혹시 `7`을 `"Bearer ".length()` 이렇게 바꿔도 될까요?
```suggestion
        return header.substring("Bearer ".length()); // "Bearer " 이후의 토큰만 추출
```